File: Web_search_job/ClientApp/src/app/models/backend/dtos/profiles/resume.dto.ts

import {UserDTO} from "../users/user.dto";
import {ActiveResumeSectionDTO} from "./active-resume-section.dto";
import {ResumeEducationDTO} from "./resume-education.dto";
import {ResumeLanguageDTO} from "./resume-language.dto";
import {ResumeLinksDTO} from "./resume-links.dto";
import {ResumePortfolioDTO} from "./resume-portfolio.dto";
import {ResumeSkillsDTO} from "./resume-skills.dto";
import {ResumeTagsDTO} from "./resume-tags.dto";
import {ResumeAboutMeDTO} from "./resume-about-me.dto";
import {ResumeHistoryWorkDTO} from "./resume-history-work.dto";



export interface ResumeDTO {
  id: number | null;
  userId: number | null;
  resumeName: string;
  resumeDescription: string | null;
  resumeEmail: string | null;
  resumePhone: string | null;
  wantedSalary: string | null;
  resumeTags: ResumeTagsDTO[];
  resumeActive: boolean;
  userInfo: UserDTO | null;
  resumeAboutMe: ResumeAboutMeDTO | null;
  activeResumeSection: ActiveResumeSectionDTO[];
  resumeEducation: ResumeEducationDTO[];
  resumeLanguage: ResumeLanguageDTO[];
  resumeLinks: ResumeLinksDTO[];
  resumePortfolio: ResumePortfolioDTO[];
  resumeSkills: ResumeSkillsDTO[];
  resumeHistoryWork: ResumeHistoryWorkDTO[];
}

File: Web_search_job/DTO/Resume/ResumeDTO.cs
using Web_search_job.DatabaseClasses.ProfileFolder;
using Web_search_job.DTO.User;

namespace Web_search_job.DTO.Resume
{
    public class ResumeDTO
    {
        public int? Id { get; set; }
        public int? UserId { get; set; }
        public string ResumeName { get; set; }
        public string? ResumeDescription { get; set; }
        public string? ResumeEmail { get; set; }
        public string? ResumePhone { get; set; }
        public string? WantedSalary { get; set; }

        public List<ResumeTagsDTO> ResumeTags { get; set; }
        public bool ResumeActive { get; set; }
        public ResumeAboutMeDTO? ResumeAboutMe { get; set; }
        public UserDTO? UserInfo { get; set; }
        public List<ActiveResumeSectionDTO>? ActiveResumeSection { get; set; }
        public List<ResumeEducationDTO>? ResumeEducation { get; set; }
        public List<ResumeLanguageDTO>? ResumeLanguage { get; set; }
        public List<ResumeLinksDTO>? ResumeLinks { get; set; }
        public List<ResumePortfolioDTO>? ResumePortfolio { get; set; }
        public List<ResumeSkillsDTO>? ResumeSkills { get; set; }
        public List<ResumeHistoryWorkDTO>? ResumeHistoryWork { get; set; }
    }
}
